### PR TITLE
Update the runtime image and deprecated pandas dataframe function

### PR DESCRIPTION
**Pull request by harshad16**
Update the runtime image and deprecated pandas dataframe function

Fixes:
- futurewarning: indexing a dataframe with a datetimelike index using a single string to slice the rows, like `frame[string]`, is deprecated and will be removed in a future version. use `frame.loc[string]` instead.

### CODE REVIEW

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/Part 2 - Data Analysis.ipynb [hunk 1]
```diff
@@ -551,7 +551,7 @@
  ],
  "metadata": {
   "kernelspec": {
-   "display_name": "Python 3 (ipykernel)",
+   "display_name": "Python 3.9.16",
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/Part 2 - Data Analysis.ipynb]: This should be left unchanged

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 2]
```diff
@@ -14,16 +14,22 @@
...
-              "dependencies": [],
-              "include_subdirectories": false
+              "kubernetes_pod_annotations": [],
+              "kubernetes_pod_labels": [],
+              "kubernetes_secrets": [],
+              "kubernetes_shared_mem_size": {},
+              "kubernetes_tolerations": [],
+              "mounted_volumes": [],
+              "filename": "load_data.ipynb",
+              "runtime_image": "quay.io/modh/runtime-images@sha256:27f12a510a034212ce4d579a970cce7aeeb33ffa32044fe88a262ae15d34e763"
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: I don't think these changes are needed, especially the fields added by what looks like our Jupyterlab instance
```
 "kubernetes_pod_annotations": [],
              "kubernetes_pod_labels": [],
              "kubernetes_secrets": [],
              "kubernetes_shared_mem_size": {},
              "kubernetes_tolerations": [],
              "mounted_volumes": [],
```

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 3]
```diff
@@ -69,14 +75,20 @@
...
-              "dependencies": [],
-              "include_subdirectories": false
+              "kubernetes_pod_annotations": [],
+              "kubernetes_pod_labels": [],
+              "kubernetes_secrets": [],
+              "kubernetes_shared_mem_size": {},
+              "kubernetes_tolerations": [],
+              "mounted_volumes": [],
+              "filename": "Part 1 - Data Cleaning.ipynb",
+              "runtime_image": "quay.io/modh/runtime-images@sha256:27f12a510a034212ce4d579a970cce7aeeb33ffa32044fe88a262ae15d34e763"
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 4]
```diff
@@ -129,12 +141,18 @@
...
-              "dependencies": [],
-              "include_subdirectories": false
+              "kubernetes_pod_annotations": [],
+              "kubernetes_pod_labels": [],
+              "kubernetes_secrets": [],
+              "kubernetes_shared_mem_size": {},
+              "kubernetes_tolerations": [],
+              "mounted_volumes": [],
+              "filename": "Part 2 - Data Analysis.ipynb",
+              "runtime_image": "quay.io/modh/runtime-images@sha256:27f12a510a034212ce4d579a970cce7aeeb33ffa32044fe88a262ae15d34e763"
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 5]
```diff
@@ -187,12 +205,18 @@
...
-              "dependencies": [],
-              "include_subdirectories": false
+              "kubernetes_pod_annotations": [],
+              "kubernetes_pod_labels": [],
+              "kubernetes_secrets": [],
+              "kubernetes_shared_mem_size": {},
+              "kubernetes_tolerations": [],
+              "mounted_volumes": [],
+              "filename": "Part 3 - Time Series Forecasting.ipynb",
+              "runtime_image": "quay.io/modh/runtime-images@sha256:27f12a510a034212ce4d579a970cce7aeeb33ffa32044fe88a262ae15d34e763"
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 6]
```diff
@@ -253,6 +277,7 @@
               "content": "Download the data",
               "associated_id_refs": [
                 {
+                  "id": "cb879ebf-6a33-4504-beb0-d78dd507e7d9",
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: what are these IDs?

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 7]
```diff
@@ -267,6 +292,7 @@
               "content": "Clean the data",
               "associated_id_refs": [
                 {
+                  "id": "363ed27d-4043-40dd-b6da-e10537d42e55",
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 8]
```diff
@@ -281,6 +307,7 @@
               "content": "Analyze the data",
               "associated_id_refs": [
                 {
+                  "id": "99b500e4-fea7-4e78-8b70-7586cd8436a7",
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 9]
```diff
@@ -295,6 +322,7 @@
               "content": "Explore approaches to predicting future temperatures",
               "associated_id_refs": [
                 {
+                  "id": "8f0fb964-c102-423a-aa93-1e019e8b7cd6",
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: same as above

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline [hunk 10]
```diff
@@ -306,7 +334,16 @@
...
+          "description": "A generic pipeline tutorial",
+          "pipeline_defaults": {
+            "kubernetes_shared_mem_size": {},
+            "kubernetes_tolerations": [],
+            "kubernetes_pod_annotations": [],
+            "kubernetes_pod_labels": [],
+            "mounted_volumes": [],
+            "env_vars": [],
+            "kubernetes_secrets": []
+          }
```
Comment (lugi0) [inline notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: I think these fields should not be included

**Review by lugi0 — CHANGES_REQUESTED** [review]

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/Part 2 - Data Analysis.ipynb (hunk 1 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/Part 2 - Data Analysis.ipynb]: ack, made the change. :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 2 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: ack, made the change. :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 4 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: ack, made the change. :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 5 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: ack, made the change. :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 7 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: Removed it :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 6 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: these are associated task ids, which are generated by elyra.
removing them as they would be re-created anyway on execution in the cluster.

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 3 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: ack, made the change. :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 8 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: Removed it :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 9 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: Removed it :+1:

File: notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline (hunk 10 above)
Comment (harshad16) [reply notebooks/500__jupyterhub/elyra/run-pipelines-on-data-science-pipelines/hello-generic-world.pipeline]: Removed it :+1:

**Comment by harshad16** [comment]
Made the changes as requested.
Ready for Review.

**Review by lugi0 — APPROVED** [review]

**Comment by lugi0** [comment]
Looks great, thanks a lot @harshad16

**Pull request merged**